SpeedrunTimer/data/tables/speedrun-timer-sct.tbm

```diff
@@ -0,0 +1,138 @@
+        self.InputOutput = require "io" --io is defaulted to the controls. Since we need it to send LiveSplit commands, we need an alias.
+        self.LSPipe = self.InputOutput.open(self.LSEndpoint, "w") --Open/start the pipe. Flush is required after every command.
+        ba.println("Speedrun Timer: Initialized - " .. tostring(time.getCurrentTime()))
+        self.Initialized = true
+    end
+
+    function SpeedrunTimer:StopTimer()
+        self.RunningStatus = false
+        ba.println("Speedrun Timer: StopTimer called - Last Segment Time: " .. self.CurrentSegmentTime)
+        self.CurrentSegmentTime = 0
+        self:PauseLiveSplit();
+    end
+
+    function SpeedrunTimer:PauseLiveSplit() --Send Pause to LiveSplit, and split if its on a tracked segment.
+        ba.println("Speedrun Timer: PauseLiveSplit called - Mission Filename: " .. mn.getMissionFilename())
+        if not(mn.isTraining() or --Training Missions
+         mn.getMissionFilename():upper() == "SM1-01" or --Start of the game.
+         self.CurrentMission == mn.getMissionFilename() --We restarted a mission.
+        ) then
+            self.LSPipe:write "split\n" --Split time.
+            self.LSPipe:flush()
+        end
+        self.LSPipe:write "pausegametime\n" --Stop the clock for the loading screen.
+        self.LSPipe:flush()
+        self.CurrentMission = mn.getMissionFilename()
+    end
+
+    function SpeedrunTimer:StartTimer()
+        self.RunningStatus = true
+        self.LastStart = time.getCurrentTime()
+        ba.println("Speedrun Timer: StartTimer called")
+        self:StartLiveSplit()
+    end
+    
+    function SpeedrunTimer:StartLiveSplit() --Send Resume to LiveSplit
+        if (self.TotalTime == 0) then
+            ba.println("Speedrun Timer: Run started, calling LiveSplit.")
+            self.LSPipe:write "start\n" --Start the game!
+            self.LSPipe:flush()
+        else
+            ba.println("Speedrun Timer: Segment Unpaused, calling LiveSplit.")
+            self.LSPipe:write "unpausegametime\n" --Loading screen is done, unpause the clock.
+            self.LSPipe:flush()
+        end
+    end
+
+    function SpeedrunTimer:ResetLiveSplit()
+        self.LSPipe:write "reset\n"
+        self.LSPipe:flush()
+        self.LSPipe:close()
+    end
+]
+
+$On Load Screen:
+[
+    if SpeedrunTimer.RunningStatus == true then
+        SpeedrunTimer:StopTimer()
+        ba.println("Speedrun Timer: Paused, Loading Screen")
+    end
+]
+
+$On Load Complete:
+[
+    if (mn == nil) then
+        return
+    end
+    if (mn.getMissionFilename():upper() == "SM3-10A") or --SM3-10A is the FSPort Credits Mission
+        (mn.getMissionFilename():upper() == "BTM-01") or --Training One in the FSPort
+        (mn.getMissionFilename():upper() == "BTM-02") or --Training Two in the FSPort
+        (mn.getMissionFilename():upper() == "BTM-03") or --Training Three in the FSPort
```

Comment:
There are two additional training missions in the FSPort, BTM-04 and BTM-05.  Just wanted to double check that they were intentionally omitted.

Reply:
Intentional. Timer starts on Commit of the first proper mission, so the timer does not need to start until then.